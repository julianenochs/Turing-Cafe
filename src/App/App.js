import React, { Component } from 'react';
import './App.css';
import ReservationsContainer from '../ReservationsContainer/ReservationsContainer';
import Form from '../Form/Form';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(response => response.json())
      .then(response => this.setState({ reservations : response }))
      .then(response => console.log('reservations', this.state.reservations))
  }
  
  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <Form />
        </div>
        <div className='resy-container'>
          <ReservationsContainer reservations={this.state.reservations}/>
        </div>
      </div>
    )
  }
}

export default App;
