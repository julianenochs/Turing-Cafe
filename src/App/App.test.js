import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('App', () => {
  const wrapper
  beforeEach(() => {
    wrapper = shallow(<App />)
  })
  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
