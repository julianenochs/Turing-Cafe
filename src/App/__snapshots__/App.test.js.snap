// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all of the data passed through 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <ReservationsContainer
      cancelReservation={[Function]}
      reservations={
        Object {
          "date": "2019/10/13",
          "name": "Juju B",
          "number": 3,
          "time": "11:00pm",
        }
      }
    />
  </div>
</div>
`;
